src/test/java/edu/carroll/dogtag/service/RegisterServiceImplTest.java

```diff
@@ -0,0 +1,521 @@
+package edu.carroll.dogtag.service;
+
+import edu.carroll.dogtag.jpa.model.Login;
+import edu.carroll.dogtag.web.form.LoginForm;
+import jakarta.transaction.Transactional;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import static org.springframework.test.util.AssertionErrors.*;
+
+/**
+ * Unit test to pass happy, crappy and crazy inputs to the RegisterService, and
+ * it should handle all situation that are tested and return the expected results.
+ * A login must be checked to verify the user had been successfully saved into the database
+ * If no user is present in the database it will always come back with null for returning
+ * a Login object. This indicates that the user was not able to register and is not in the
+ * database
+ */
+@SpringBootTest
+@Transactional
+class RegisterServiceImplTest {
+    private static final String username = "usertest";
+    private static final String usernameShort = "uShrt";
+    private static final String password = "passwordtest";
+    private static final String passwordShort = "pShort";
+    private static final String email = "emailTest@gmail.com";
+    private static final String emailInvalid = "emailInvalid";
+    private static final Logger log = LoggerFactory.getLogger(RegisterServiceImplTest.class);
```

Comment:
Unit tests should not use any log messages in them.

[ This was discussed during the demo. ]